### DISCUSSION TITLE
能做到按使用量计费的 FaaS 吗？

### DISCUSSION THREAD

**Discussion by linonetwo** [WasmEdge/WasmEdge#1547] [Q&A] [answered]
有没有可能和Web3.0结合起来

我们之前就想让去中心化语义网 BaaS https://github.com/solid/node-solid-server 能有按使用量计费的 FaaS 能力，这样 BaaS 的用户就能用 FaaS 运行全盘搜索等能力。希望能像以太坊一样，占用多少服务器资源就收多少钱。

SoLiD 自带 DID 身份服务，不过还需要一个类似 https://www.huaweicloud.com/product/apig.html 的计费收费网关挡在 FaaS 前面…

**Comment by alabulei1** [comment, marked as answer]
Hi @linonetwo ，谢谢你的问题。WasmEdge 有 gas 机制，可以做到的。https://wasmedge.org/book/en/#with-gas-limit-enabled

可以试一下呀，如有问题，随时联系呀。
